### PR TITLE
12249 - Conditional Payloads

### PR DESCRIPTION
**Pull request by seaders**

### CODE REVIEW

**Comment by adam-govan** [comment]
@seaders  Where do you update the Campaign API Version?

File: unity3d/Assets/Swrve/SwrveSDK/SwrveSDK.cs [hunk 1]
```diff
@@ -1323,20 +1323,20 @@ public string GetAppStoreLink (int gameId)
     /// <returns>
     /// In-app message for the given event.
     /// </returns>
-	public SwrveMessage GetMessageForEvent (string eventName) {
-		if (!checkCampaignRules (eventName, SwrveHelper.GetNow())) {
-			return null;
-		}
+	  public SwrveMessage GetMessageForEvent (string eventName, IDictionary<string, string> payload=null) {
```
Comment (dominicmarmionswrve) [inline unity3d/Assets/Swrve/SwrveSDK/SwrveSDK.cs:1326]: Excuse my lack of unity knowledge here, but are we changing the signature of a public method here? Do our customers use this method?

Comment (seaders) [inline unity3d/Assets/Swrve/SwrveSDK/SwrveSDK.cs:1326]: It's not changing the signature, it's adding a new, default argument.  People can still call it about `GetMessageForEvent("asdf")`, but they (and we) can also call `GetMessageForEvent("asdf", new Dictionary<string, string>{ ... })`.

Comment (dominicmarmionswrve) [inline unity3d/Assets/Swrve/SwrveSDK/SwrveSDK.cs:1326]: cool! 👍

File: unity3d/Assets/Swrve/SwrveSDK/SwrveSDKImp.cs [hunk 2]
```diff
@@ -89,8 +89,8 @@ public partial class SwrveSDK
     protected bool campaignsAndResourcesInitialized;
 
     // Talk related
-    private static readonly int CampaignAPIVersion = 1;
-    private static readonly int CampaignEndpointVersion = 4;
+    private static readonly int CampaignEndpointVersion = 6;
```
Comment (seaders) [inline unity3d/Assets/Swrve/SwrveSDK/SwrveSDKImp.cs:92]: @adam-govan here.  Here is where I do that thing you mentioned.  Riiiiight here.

Comment (adam-govan) [inline unity3d/Assets/Swrve/SwrveSDK/SwrveSDKImp.cs:92]: Lovely 🚀  💯

**Comment by adam-govan** [comment]
🚀

File: unity3d/Assets/Swrve/SwrveSDK/Messaging/SwrveConditions.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            if (string.IsNullOrEmpty (key) || string.IsNullOrEmpty (value)) {
+                return null;
+            }
+            return new SwrveConditions (TriggerOperatorType.EQUALS, key, value);
+        } else if (isRoot && (op == OP_AND_KEY)) {
+            IList<object> jsonArgs = (IList<object>)json [ARGS_KEY];
+            List<SwrveConditions> args = new List<SwrveConditions> ();
+            var it = jsonArgs.GetEnumerator ();
+            while (it.MoveNext ()) {
+                SwrveConditions condition = LoadFromJson ((Dictionary<string, object>)it.Current, false);
```
Comment (dominicmarmionswrve) [inline unity3d/Assets/Swrve/SwrveSDK/Messaging/SwrveConditions.cs:106]: Just seen this is recursive. 😞

**Pull request merged**